rustls/examples/internal/bogo_shim.rs

```diff
@@ -185,22 +185,19 @@ impl rustls::ClientCertVerifier for DummyClientAuth {
         true
     }
 
-    fn client_auth_mandatory(&self, _sni: Option<&webpki::DnsName>) -> Option<bool> {
+    fn client_auth_mandatory(&self, _sni: Option<&str>) -> Option<bool> {
```

Comment:
Why not use the new `DnsName` type that you created in an earlier commit here? That way, the fact that `sni` has been checked to be syntactically valid is encoded in the type.

Reply:
> Why not use the new `DnsName` type that you created in an earlier commit here? That way, the fact that `sni` has been checked to be syntactically valid is encoded in the type.

Not just here, but elsewhere.

Also, I think the way this API uses `sni` is not useful as explained in https://github.com/ctz/rustls/issues/473. So, maybe we should come up with a plan for 473 and see how we can remove all these `sni` uses.